### PR TITLE
Components endpoint now returns valid json

### LINKED ISSUE #339: :bug: [BUG]: Ouranos PAVICS node error on `/components` reporting
## Summary

The JSON contents of `https://pavics.ouranos.ca/components/` are invalid. They are missing quotes and `,` for ` ./config/ncwms2`. 

## Details

```json
{"components": ["bird-house/birdhouse-deploy:config/proxy",
"bird-house/birdhouse-deploy:config/canarie-api",
"bird-house/birdhouse-deploy:config/postgres",
"bird-house/birdhouse-deploy:config/wps_outputs-volume",
"bird-house/birdhouse-deploy:config/data-volume",
"bird-house/birdhouse-deploy:config/malleefowl",
"bird-house/birdhouse-deploy:config/flyingpigeon",
"bird-house/birdhouse-deploy:config/catalog",
"bird-house/birdhouse-deploy:config/mongodb",
"bird-house/birdhouse-deploy:config/phoenix",
"bird-house/birdhouse-deploy:config/geoserver",
"bird-house/birdhouse-deploy:config/finch",
"bird-house/birdhouse-deploy:config/raven",
"bird-house/birdhouse-deploy:config/hummingbird",
"bird-house/birdhouse-deploy:config/thredds",
"bird-house/birdhouse-deploy:config/portainer",
"bird-house/birdhouse-deploy:config/magpie",
"bird-house/birdhouse-deploy:config/twitcher",
"bird-house/birdhouse-deploy:config/jupyterhub",
        ./config/ncwms2
"bird-house/birdhouse-deploy:config/frontend",
"bird-house/birdhouse-deploy:config/project-api",
"bird-house/birdhouse-deploy:config/solr",
"bird-house/birdhouse-deploy:optional-components/canarie-api-full-monitoring",
"bird-house/birdhouse-deploy:optional-components/wps-healthchecks",
"bird-house/birdhouse-deploy:optional-components/testthredds",
"bird-house/birdhouse-deploy:optional-components/generic_bird",
"bird-house/birdhouse-deploy:components/weaver",
"bird-house/birdhouse-deploy:components/scheduler",
"bird-house/birdhouse-deploy:components/monitoring"]}
```

### To Reproduce

1. GET request on `https://pavics.ouranos.ca/components/`

## Environment

| Information           | Value
| --------------------- | --------------------------------------------------------------
| Server/Platform URL   | https://pavics.ouranos.ca
| Version Tag/Commit    | (?) latest on 2023-06-05
| Related issues/PR     | n/a
| Related components    | platform
| Custom configuration  | (?) n/a

## Concerned Organizations

@tlvu

### PR DESCRIPTION
**Pull request by mishaschwartz**
## Overview

The JSON string reported by the `/components/` path was not valid JSON due to a misconfigured regular expression used to generate the content. The issue was that integers were not being properly parsed by the regular expression meaning that paths that contained integers other than 0 were not recognized as valid paths.

## Changes

**Non-breaking changes**

None: bugfix

**Breaking changes**

None

## Related Issue / Discussion

- Resolves #339

### CODE REVIEW

File: birdhouse/scripts/get-components-json.include.sh [hunk 1]
```diff
@@ -57,7 +57,7 @@ BIRDHOUSE_DEPLOY_COMPONENTS_BASE="bird-house/birdhouse-deploy:"
 BIRDHOUSE_DEPLOY_COMPONENTS_LIST=$( \
   echo "${BIRDHOUSE_DEPLOY_COMPONENTS_LIST_ACTIVE}" \
   | grep "${BIRDHOUSE_DEPLOY_COMPONENTS_LIST_KNOWN}" \
-  | sed -E 's|^\s*([A-Za-z0-0./_-]+)\s*$|"\1",|g' \
+  | sed -E 's|^\s*([A-Za-z0-9./_-]+)\s*$|"\1",|g' \
```
Comment (tlvu) [inline birdhouse/scripts/get-components-json.include.sh:60]: Oh geez !!!  But then how on earth did this work when `./pavics-compose.sh` is called manually and not when it is called by autodeploy !

**Comment by tlvu** [comment]
Oh sorry, this fixes the badly formatted json, not the empty string when called by autodeploy.  I was mixed up.

**Comment by mishaschwartz** [comment]
@fmigneault  can I merge this one?

File: birdhouse/scripts/get-components-json.include.sh (hunk 1 above)
Comment (fmigneault) [reply birdhouse/scripts/get-components-json.include.sh:62]: All the services except `ncWMS2` are alpha-only, not alphanumeric!  Could it be that `autodeploy` does not use it?
Silly mistake. 😅 Nice find.

**Review by fmigneault — APPROVED** [review]

**Pull request merged**